### PR TITLE
Implement book search functionality with /api/books/search endpoint

### PR DESCRIPTION
**Pull request by ShastkivRuslan**
Implement book search functionality with /api/books/search endpoint.
- Added /api/books/search endpoint to retrieve books based on search parameters.
- Created BookSearchParametersDto with fields for title, author, and price.
- Implemented logic to filter books based on provided search criteria.
Note: 
- Added logic to search for books based on price.
- If one price parameter is provided, filter for books with a price greater than or equal to the specified value.
- If two price parameters are provided, filter for books within the specified price range.

### CODE REVIEW

File: src/main/java/ruslan/shastkiv/bookstore/controller/BookController.java [hunk 1]
```diff
@@ -47,4 +48,9 @@ public BookDto updateBook(@PathVariable Long id, @RequestBody CreateBookRequestD
     public void deleteBook(@PathVariable Long id) {
         bookService.deleteBook(id);
     }
+
+    @GetMapping("/search")
+    public List<BookDto> search(BookSearchParametersDto searchParametersDto) {
```
Comment (Elena-Bruyako) [inline src/main/java/ruslan/shastkiv/bookstore/controller/BookController.java:53]: you can add pageable here

File: src/main/java/ruslan/shastkiv/bookstore/repository/book/specification/AuthorSpecificationProvider.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+import java.util.Arrays;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+import ruslan.shastkiv.bookstore.model.Book;
+import ruslan.shastkiv.bookstore.repository.SpecificationProvider;
+
+@Component
+public class AuthorSpecificationProvider implements SpecificationProvider<Book> {
+    private static final String KEY = "author";
```
Comment (Elena-Bruyako) [inline src/main/java/ruslan/shastkiv/bookstore/repository/book/specification/AuthorSpecificationProvider.java]: let's be more specify AUTHOR_COLUMN for example
Check other places

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/ruslan/shastkiv/bookstore/controller/BookController.java (hunk 1 above)
Comment (ShastkivRuslan) [reply src/main/java/ruslan/shastkiv/bookstore/controller/BookController.java:53]: > you can add pageable here

If you mean pagination, I'm on my way to learning about it. Once I cover that topic, I'll definitely add it here.

**Review by Elena-Bruyako — APPROVED** [review]
well done

**Pull request merged**